solidity/contracts/converter/ConverterBase.sol

```diff
@@ -231,26 +213,51 @@ abstract contract ConverterBase is ConverterVersion, IConverter, TokenHolder, Co
     }
 
     /**
-     * @dev withdraws tokens held by the converter and sends them to an account
+     * @dev transfers reserves to a new converter during an upgrade
+     * can only be called by the converter upgraded which should be set at its owner
+     *
+     * @param _newConverter account to receive the new amount
+     */
+    function upgradeReserves(address _newConverter) external override protected ownerOnly {
+        // verify that the converter is inactive or that the owner is the upgrader contract
+        require(owner == addressOf(CONVERTER_UPGRADER), "ERR_ACCESS_DENIED");
+
+        uint256 reserveCount = reserveTokens.length;
+        for (uint256 i = 0; i < reserveCount; ++i) {
+            IERC20 reserveToken = reserveTokens[i];
+
+            uint256 amount;
+            if (reserveToken == NATIVE_TOKEN_ADDRESS) {
+                amount = address(this).balance;
+            } else {
+                amount = reserveToken.balanceOf(address(this));
+            }
+
+            safeTransfer(reserveToken, _newConverter, amount);
+
+            syncReserveBalance(reserveToken);
+        }
+    }
+
+    /**
+     * @dev withdraws non-reserve or inactive tokens held by the converter and sends them to an account
      * can only be called by the owner
      * note that reserve tokens can only be withdrawn by the owner while the converter is inactive
      * unless the owner is the converter upgrader contract
      *
-     * @param _token   ERC20 token contract address
-     * @param _to      account to receive the new amount
+     * @param _token ERC20 token contract address
+     * @param _to account to receive the new amount
      * @param _amount  amount to withdraw
      */
-    function withdrawTokens(
+    function withdrawInactiveTokens(
```

Comment:
I would completely remove the option to withdraw a non-reserve token, owners practically never use it and it just adds complexity to a sensitive code.

Reply:
Roger 👍

Reply:
Questions that should have been asked regardless of this change:
1. Why should an owner who is not the converter upgrader have the capability of withdrawing?
2. Why do we need the input amount here? I assume that this function should always withdraw everything, whether it's upon converter upgrade or due to some other reason (for which the owner needs to withdraw).
3. Why do we need the input token here? I assume that this function should always withdraw everything, whether it's upon converter upgrade or due to some other reason (for which the owner needs to withdraw).

Reply:
My question answered.